src/providers/ogr/qgsogrdataitems.cpp

```diff
@@ -110,6 +114,38 @@ QString QgsOgrLayerItem::layerName() const
     return info.completeBaseName();
 }
 
+#ifdef HAVE_GUI
+QList<QAction *> QgsOgrLayerItem::actions()
+{
+  QList<QAction *> lst;
+  QAction *actionDeleteLayer = new QAction( tr( "Delete %1" ).arg( mName ), this );
+  connect( actionDeleteLayer, &QAction::triggered, this, &QgsOgrLayerItem::deleteLayer );
+  lst.append( actionDeleteLayer );
+  return lst;
+}
+
+void QgsOgrLayerItem::deleteLayer()
+{
+  if ( QMessageBox::question( nullptr, QObject::tr( "Delete Layer" ),
+                              QObject::tr( "Are you sure you want to delete %1?" ).arg( mName ),
+                              QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) != QMessageBox::Yes )
+    return;
+
+  QString errCause;
+  bool res = ::deleteLayer( mUri, errCause );
+  if ( !res )
+  {
+    QMessageBox::warning( nullptr, tr( "Delete Layer" ), errCause );
+  }
+  else
+  {
+    QMessageBox::information( nullptr, tr( "Delete Layer" ), tr( "Layer deleted successfully." ) );
```

Comment:
I don't know, but if you are suggesting to show the message inside the browser's widget I'm afraid this is not a good idea:
- the browser widget could not be large enough (the one in the left is typically very narrow)
- I don't think that showing flash messages in different parts of the GUI is a good design: users quickly become used to see messages in the main canvas are and they get easily confused when the message appears for example in the auth manager window widget.